packages/node/src/integrations/tracing/koa.ts

```diff
@@ -1,14 +1,28 @@
 import { registerInstrumentations } from '@opentelemetry/instrumentation';
 import { KoaInstrumentation } from '@opentelemetry/instrumentation-koa';
-import { captureException, defineIntegration } from '@sentry/core';
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions';
+import { captureException, defineIntegration, getCurrentScope, spanToJSON } from '@sentry/core';
 import type { IntegrationFn } from '@sentry/types';
 
 const _koaIntegration = (() => {
   return {
     name: 'Koa',
     setupOnce() {
       registerInstrumentations({
-        instrumentations: [new KoaInstrumentation()],
+        instrumentations: [
+          new KoaInstrumentation({
+            requestHook(span, info) {
+              if (span.isRecording() && info.layerType === 'router') {
```

Comment:
l: do we need this check? Can we not just directly check if the HTTP_ROUTE attribute exist, and if so use it?

Reply:
I guess we can 🤔 I wanted to play it safe. q: is it safe to call `spanToJSON` on a non-recording span?

Reply:
yes, it will just return `{}` then, basically!